testing/spectest/shared/altair/light_client/single_merkle_proof.go

```diff
@@ -0,0 +1,99 @@
+	"github.com/golang/snappy"
+)
+
+// RunLightClientSingleMerkleProofTests executes "light_client/single_merkle_proof/{BeaconState}" tests.
+func RunLightClientSingleMerkleProofTests(t *testing.T, config string) {
+	require.NoError(t, utils.SetConfig(t, config))
+
+	_, testsFolderPath := utils.TestFolders(t, config, "altair", "light_client/single_merkle_proof")
+	testTypes, err := util.BazelListDirectories(testsFolderPath)
+	require.NoError(t, err)
+
+	if len(testTypes) == 0 {
+		t.Fatalf("No test types found for %s", testsFolderPath)
+	}
+
+	for _, testType := range testTypes {
+		testFolders, testsFolderPath := utils.TestFolders(t, config, "altair", fmt.Sprintf("light_client/single_merkle_proof/%s", testType))
+		for _, folder := range testFolders {
+			helpers.ClearCache()
+			t.Run(fmt.Sprintf("%v/%v", testType, folder.Name()), func(t *testing.T) {
+				folderPath := path.Join(testsFolderPath, folder.Name())
+				runLightClientSingleMerkleProofTest(t, folderPath, folder.Name())
+			})
+		}
+	}
+}
+
+func runLightClientSingleMerkleProofTest(t *testing.T, testFolderPath string, testName string) {
+	ctx := context.Background()
+
+	beaconStateFile, err := util.BazelFileBytes(path.Join(testFolderPath, "object.ssz_snappy"))
+	require.NoError(t, err)
+	beaconStateSSZ, err := snappy.Decode(nil, beaconStateFile)
+	require.NoError(t, err, "Failed to decompress")
+	beaconStateBase := &ethpb.BeaconStateAltair{}
+	require.NoError(t, beaconStateBase.UnmarshalSSZ(beaconStateSSZ), "Failed to unmarshal")
+	beaconState, err := state_native.InitializeFromProtoAltair(beaconStateBase)
+	require.NoError(t, err)
+	beaconStateRoot, err := beaconState.HashTreeRoot(ctx)
+	require.NoError(t, err)
+	type Proof struct {
+		Leaf      string   `json:"leaf"`
+		LeafIndex uint64   `json:"leaf_index"`
+		Branch    []string `json:"branch"`
+	}
+	proofFile, err := util.BazelFileBytes(path.Join(testFolderPath, "proof.yaml"))
+	require.NoError(t, err)
+	var proof Proof
+	require.NoError(t, utils.UnmarshalYaml(proofFile, &proof))
+	leaf, err := hex.DecodeString(proof.Leaf[2:])
+	if err != nil {
+		fmt.Printf("Error decoding leaf: %v\n", err)
+	}
+	require.NoError(t, err)
+	var branch [][]byte
+	for _, b := range proof.Branch {
+		bBytes, err := hex.DecodeString(b[2:])
+		require.NoError(t, err)
+		branch = append(branch, bBytes)
+	}
+
+	var item []byte
+	if strings.Contains(testName, "current_sync_committee") {
+		syncCommittee, err := beaconState.CurrentSyncCommittee()
+		require.NoError(t, err)
+		item32, err := syncCommittee.HashTreeRoot()
+		require.NoError(t, err)
+		item = item32[:]
+	} else if strings.Contains(testName, "next_sync_committee") {
+		syncCommittee, err := beaconState.NextSyncCommittee()
+		require.NoError(t, err)
+		item32, err := syncCommittee.HashTreeRoot()
+		require.NoError(t, err)
+		item = item32[:]
+	} else if strings.Contains(testName, "finality_root") {
+		item = beaconState.FinalizedCheckpoint().Root
+	}
```

Comment:
Should you handle the case where the testName is none of the above?

```suggestion
	} else if strings.Contains(testName, "finality_root") {
		item = beaconState.FinalizedCheckpoint().Root
	} else {
		t.Fatalf("Unknown test type %s", testName)
	}
```

I believe the DeepEqual will fail below but this would be a better feedback to the user imo

Reply:
I'm not sure about this. do we want our tests to fail if a new spec test type gets added here? or should we just do sth like: 
```go
else {
    return
}
```
because if we do what you suggest, then we have to add support for new spec tests immediately.

Reply:
Yes - we should fail if spectests introduces a new type and we haven't added a handler for it.